File: src/main/java/com/lcaohoanq/nocket/domain/wallet/IWalletService.java
package com.lcaohoanq.nocket.domain.wallet;

import com.lcaohoanq.nocket.domain.wallet.WalletDTO.WalletResponse;
import java.util.UUID;

public interface IWalletService {

    WalletResponse getByUserId(UUID userId);
    void updateAccountBalance(UUID userId, Long payment) throws Exception;
    
}

File: server/src/main/kotlin/com/lcaohoanq/nocket/domain/wallet/IWalletService.kt
package com.lcaohoanq.nocket.domain.wallet

import com.lcaohoanq.nocket.domain.wallet.WalletPort.WalletResponse
import jakarta.mail.MessagingException
import org.springframework.retry.annotation.Backoff
import org.springframework.retry.annotation.Retryable
import java.util.*

interface IWalletService {
    fun getByUserId(userId: UUID): WalletResponse
     
    @Retryable(
        retryFor = [MessagingException::class],
        maxAttempts = 3,
        backoff = Backoff(delay = 2000)
    )
    @Throws(Exception::class)
    fun updateAccountBalance(userId: UUID, payment: Long)
}
